hail/python/test/hail/methods/test_qc.py

```diff
@@ -285,3 +288,47 @@ def test_charr(self):
 
         assert pytest.approx(d['C1046::HG02024'], abs=0.0001) == .00126
         assert pytest.approx(d['C1046::HG02025'], abs=0.0001) == .00124
+
+    @skip_unless_service_backend()
+    @set_gcs_requester_pays_configuration(GCS_REQUESTER_PAYS_PROJECT)
+    def test_vep_grch37_consequence_true(self):
+        gnomad_vep_result = hl.import_vcf(resource('sample.gnomad.exomes.r2.1.1.sites.chr1.vcf.gz'), reference_genome='GRCh37', force=True)
+        hail_vep_result = hl.vep(gnomad_vep_result, csq=True)
+
+        assert gnomad_vep_result.select_rows(vep=gnomad_vep_result.info.vep.map(lambda x: x.split('|')[:8])).rows()._same(
+            hail_vep_result.select_rows(vep=hail_vep_result.vep.map(lambda x: x.split('|')[:8])).rows().drop('vep_csq_header'))
+
+        vep_csq_header = hl.eval(hail_vep_result.vep_csq_header)
+        assert 'Consequence annotations from Ensembl VEP' in vep_csq_header, vep_csq_header
+
+    @skip_unless_service_backend()
+    @set_gcs_requester_pays_configuration(GCS_REQUESTER_PAYS_PROJECT)
+    def test_vep_grch38_consequence_true(self):
+        gnomad_vep_result = hl.import_vcf(resource('sample.gnomad.genomes.r3.0.sites.chr1.vcf.gz'), reference_genome='GRCh38', force=True)
+        hail_vep_result = hl.vep(gnomad_vep_result, csq=True)
+
+        assert gnomad_vep_result.select_rows(vep=gnomad_vep_result.info.vep.map(lambda x: x.split('|')[:8])).rows()._same(
+            hail_vep_result.select_rows(vep=hail_vep_result.vep.map(lambda x: x.split('|')[:8])).rows().drop('vep_csq_header'))
+
+        vep_csq_header = hl.eval(hail_vep_result.vep_csq_header)
+        assert 'Consequence annotations from Ensembl VEP' in vep_csq_header, vep_csq_header
+
+    @skip_unless_service_backend()
+    @set_gcs_requester_pays_configuration(GCS_REQUESTER_PAYS_PROJECT)
+    def test_vep_grch37_consequence_false(self):
+        mt = hl.import_vcf(resource('sample.gnomad.exomes.r2.1.1.sites.chr1.vcf.gz'), reference_genome='GRCh37', force=True)
+        hail_vep_result = hl.vep(mt, csq=False)
+        ht = hail_vep_result.rows()
+        ht = ht.select(variant_class=ht.vep.variant_class)
+        result = ht.head(1).collect()[0]
+        assert result.variant_class == 'SNV', result
```

Comment:
We need to verify more than just this. At the very least we should write out the expected results as another file, manually inspect that it looks good, check that in, and assert `result._same(checked_in_results)`.

Reply:
My reasoning for this test being so simple was it's more to make sure the schemas are correct when csq=True. The other tests with csq=False tests the output is correct with the existing gnomAD data. If you feel we need both tests to test the actual VEP functionality then I can go ahead and add a file with whatever the current output is.